src/devices/s3318p.c

```diff
@@ -0,0 +1,149 @@
+#include "rtl_433.h"
+#include "data.h"
+#include "util.h"
+
+/* Conrad Electronics S3318P outdoor sensor
+ *
+ * Transmit Interval: every ~50s
+ * Message Format: 40 bits (10 nibbles)
+ *
+ *
+ * Nibble:    1   2    3   4    5   6    7   8    9   10
+ * Type:   PP IIIIIIII ??CCTTTT TTTTTTTT HHHHHHHH XB?????? PP
+ * BIT/8   00 01234567 01234567 01234567 01234567 01234567 00
+ * BIT/A   00 01234567 89012345 57890123 45678901 23456789 00
+ *            0          1          2          3
+ * I = sensor ID (changes on battery change)
+ * C = channel number
+ * T = temperature
+ * H = humidity
+ * X = tx-button pressed
+ * B = low battery
+ * P = Pre-/Postamble
+ * ? = unknown meaning
+ *
+ *
+ * [01] {42} 04 15 66 e2 a1 00 : 00000100 00010101 01100110 11100010 10100001 00 ---> Temp/Hum/Ch:23.2/46/1
+ *
+ * Temperature:
+ * Sensor sends data in °F, lowest supported value is 90°F
+ * 12 bit uingned and scaled by 10 (Nibbles: 6,5,4)
+ * in this case "011001100101" =  1637/10 - 90 = 73.7 °F (23.17 °C)
+ *
+ * Humidity:
+ * 8 bit unsigned (Nibbles 8,7)
+ * in this case "00101110" = 46
+ *
+ * Channel number: (Bits 10,11) + 1
+ * in this case "00" --> "00" +1 = Channel1
+ *
+ * Battery status: (Bit 33) (0 normal, 1 voltage is below ~2.7 V)
+ * TX-Button: (Bit 32) (0 indicates regular transmission, 1 indicates requested by pushbutton)
+ *
+ * Rolling Code / Device ID: (Nibble 1)
+ * changes on every battery change
+ *
+ * Unknown1: (Bits 8,9) changes not so often
+ * Unknown2: (Bits 36-39) changes with every packet, probably checksum
+ * Unknown3: (Bits 34,35) changes not so often, mayby also part of the checksum
+ *
+ */
+
+
+static int s3318p_callback(bitbuffer_t *bitbuffer) {
+    bitrow_t *bb = bitbuffer->bb;
+    data_t *data;
+    char time_str[LOCAL_TIME_BUFLEN];
+
+    /* Get time now */
+    local_time_str(0, time_str);
+
+    /* Reject codes of wrong length */
+    if ( 42 != bitbuffer->bits_per_row[1])
+      return 0;
+
+    /* shift all the bits left 2 to align the fields */
+    int i;
+    for (i = 0; i < BITBUF_COLS-1; i++) {
+      uint8_t bits1 = bb[1][i] << 2;
+      uint8_t bits2 = (bb[1][i+1] & 0xC0) >> 6;
+      bits1 |= bits2;
+      bb[1][i] = bits1;
+    }
+
+    uint8_t humidity;
+    uint8_t button;
+    uint8_t battery_low;
+    uint8_t channel;
+    uint8_t sensor_id;
+    uint16_t temperature_with_offset;
+    float temperature_f;
+    float temperature_c;
+
+    /* IIIIIIII ??CCTTTT TTTTTTTT HHHHHHHH XB?????? PP */
+    humidity = (uint8_t)(((bb[1][3] & 0x0F) << 4) | ((bb[1][3] & 0xF0) >> 4));
+    button = (uint8_t)(bb[1][4] >> 7);
+    battery_low = (uint8_t)((bb[1][4] & 0x40) >> 6);
+    channel = (uint8_t)(((bb[1][1] & 0x30) >> 4) + 1);
+    sensor_id = (uint8_t)(bb[1][0]);
+
+    temperature_with_offset = (uint16_t)(((bb[1][2] & 0x0F) << 8) | (bb[1][2] & 0xF0) | (bb[1][1] & 0x0F));
+    temperature_f = (float)((temperature_with_offset - 900) / 10.0);
+    temperature_c = fahrenheit2celsius(temperature_f);
+
```

Comment:
You are supposed to report the data from the sensor as you got it. So remove this line and change the output to temperature_f instead. There is a command line option that will convert the data properly for you.